js/base/Exchange.js

```diff
@@ -2397,6 +2396,19 @@ module.exports = class Exchange {
         return interest;
     }
 
+    async fetchFundingRate (symbol, params = {}) {
+        if (this.has['fetchFundingRates']) {
+            const market = await this.market (symbol);
+            if (!market['contract']) {
+                throw new BadSymbol (this.id + ' fetchFundingRate () supports contract markets only');
+            }
+            const rates = await this.fetchFundingRates ([ symbol ], params);
+            return this.safeValue (rates, symbol);
```

Comment:
I'd suggest that we throw an exception here if the resulting value is not found in the response, otherwise we return an undocumented `undefined` value when the `symbol` is missing in `rates`. In general, we should either return a unified structure, or throw an exception.